extensions/ql-vscode/src/databases/db-module.ts

```diff
@@ -0,0 +1,43 @@
+import * as vscode from 'vscode';
+import { isNewQueryRunExperienceEnabled } from '../config';
+import { logger } from '../logging';
+import { DisposableObject } from '../pure/disposable-object';
+import { DbConfigStore } from './db-config-store';
+import { DbManager } from './db-manager';
+import { DbPanel } from './ui/db-panel';
+
+export class DbModule extends DisposableObject {
+  public async initialize(
+    extensionContext: vscode.ExtensionContext
+  ): Promise<void> {
+    if (extensionContext.extensionMode !== vscode.ExtensionMode.Development ||
+      !isNewQueryRunExperienceEnabled()) {
+      // Currently, we only want to expose the new database panel when we
+      // are in development mode and the developer has enabled the new 
+      // query run experience.
```

Comment:
Do we also need to check `canary` here? 🤔 
https://github.com/github/vscode-codeql/blob/26e2021551d34a00fab3865147435f06929d910a/extensions/ql-vscode/package.json#L1213-L1215
(That's how we define the panel in the package.json, but I'm not sure how that interacts with the `DbModule` here.)

Reply:
Good shout, thanks!